crates/polars-plan/src/dsl/function_expr/search_sorted.rs

```diff
@@ -4,11 +4,13 @@ pub(super) fn search_sorted_impl(s: &mut [Column], side: SearchSortedSide) -> Po
     let sorted_array = &s[0];
     let search_value = &s[1];
 
+    let sorted_series = sorted_array.as_materialized_series();
+    let descending = matches!(sorted_series.is_sorted_flag(), IsSorted::Descending);
```

Comment:
I should have probably noticed this earlier, but this cannot be. This needs to be a flag given. You cannot rely on the `IsSorted` flag being set or not set. If people are reading a file and know themselves it is sorted, they should not need to sort beforehand or set the sorted flag.

We need a flag like this:

`.search_sorted(descending=True/False)`

Reply:
Sounds like that doesn't make sense since the flag may e.g. be chunk-specific state, not global Series state, and so can't be trusted.

Reply:
This will probably be big enough change that I should split this PR in two.

Reply:
Moved into https://github.com/pola-rs/polars/pull/22825

Reply:
Done separately.